### PR TITLE
Fix links in documentation

### PR DESCRIPTION
**Pull request by turbolent**
## Description

The Developer Portal now requires relative paths or full URLs, absolute paths are not supported anymore.
This is checked by the new CI job, "Developer Portal Content Check".

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by turbolent** [comment]
@joshuahannan @satyamakgec It looks like two images in the tutorials are broken: https://github.com/onflow/cadence/pull/2142/checks?check_run_id=9514273337

**Comment by turbolent** [comment]
@bthaile it looks like relative paths are not fully supported, `..` does not seem to work

**Review by dsainati1 — APPROVED** [review]

**Review by SupunS — APPROVED** [review]

**Comment by bthaile** [comment]
I'm confused on how links should work. It has been said that if the links work in github they should work on the portal. 
Its not intuitive how links work.
cuz the first link changed in the PR works fine in the dev portal. 
In docs/design-patterns.mdx file, 
"/cadence/language/capability-based-access-control" resolves fine. But in github the link is broken.

I'm curious if the link was changed to work in github would it break on the dev portal?

In docs/design-patterns.mdx file, change
"/cadence/language/capability-based-access-control" to 
"docs/language/capability-based-access-control.md"

If not there is a redirects section in the flow-docs.json where urls can be redirected.
https://github.com/onflow/cadence/blob/8b67509f38aff492b2e175a8a29ccf97fe83c65a/docs/flow-docs.json#L44

File: docs/tutorial/01-first-steps.mdx [hunk 1]
```diff
@@ -14,11 +14,11 @@ Cadence introduces new features to smart contract programming that help develope
 - Type safety and a strong static type system
 - Resource-oriented programming, a new paradigm that pairs linear types with object capabilities to create a secure and declarative model for digital ownership
   by ensuring that resources (and their associated assets) can only exist in one location at a time, cannot be copied, and cannot be accidentally lost or deleted
-- Built-in pre-conditions and post-conditions for functions and [transactions](/cadence/language/transactions)
+- Built-in pre-conditions and post-conditions for functions and [transactions](../language/transactions)
 - The utilization of capability-based security, which enforces access control by requiring that access to objects
   is restricted to only the owner and those who have a valid reference to the object
 
-Please see the [Cadence introduction](/cadence) for more information about the high level design of the language.
+Please see the [Cadence introduction](..) for more information about the high level design of the language.
```
Comment (turbolent) [inline docs/tutorial/01-first-steps.mdx:21]:
```suggestion
Please see the [Cadence introduction](../) for more information about the high level design of the language.
```

File: docs/tutorial/02-hello-world.mdx [hunk 2]
```diff
@@ -51,7 +51,7 @@ without the need for any trusted third party anywhere in the process, because th
 Programs that run on blockchains are commonly referred to as smart contracts
 because they mediate important functionality (such as currency) without having to rely on a central authority (like a bank).
 
-[Cadence is the resource-oriented programming language](/cadence)
+[Cadence is the resource-oriented programming language](..)
```
Comment (turbolent) [inline docs/tutorial/02-hello-world.mdx:54]:
```suggestion
[Cadence is the resource-oriented programming language](../)
```

**Review by turbolent — COMMENTED** [review]
Trying `../` instead of `..` ...

File: docs/tutorial/01-first-steps.mdx [hunk 3]
```diff
@@ -14,11 +14,11 @@ Cadence introduces new features to smart contract programming that help develope
 - Type safety and a strong static type system
 - Resource-oriented programming, a new paradigm that pairs linear types with object capabilities to create a secure and declarative model for digital ownership
   by ensuring that resources (and their associated assets) can only exist in one location at a time, cannot be copied, and cannot be accidentally lost or deleted
-- Built-in pre-conditions and post-conditions for functions and [transactions](/cadence/language/transactions)
+- Built-in pre-conditions and post-conditions for functions and [transactions](../language/transactions)
 - The utilization of capability-based security, which enforces access control by requiring that access to objects
   is restricted to only the owner and those who have a valid reference to the object
 
-Please see the [Cadence introduction](/cadence) for more information about the high level design of the language.
+Please see the [Cadence introduction](../) for more information about the high level design of the language.
```
Comment (turbolent) [inline docs/tutorial/01-first-steps.mdx:21]:
```suggestion
Please see the [Cadence introduction](https://developers.flow.com/cadence) for more information about the high level design of the language.
```

File: docs/tutorial/02-hello-world.mdx [hunk 4]
```diff
@@ -51,7 +51,7 @@ without the need for any trusted third party anywhere in the process, because th
 Programs that run on blockchains are commonly referred to as smart contracts
 because they mediate important functionality (such as currency) without having to rely on a central authority (like a bank).
 
-[Cadence is the resource-oriented programming language](/cadence)
+[Cadence is the resource-oriented programming language](../)
```
Comment (turbolent) [inline docs/tutorial/02-hello-world.mdx:54]:
```suggestion
[Cadence is the resource-oriented programming language](https://developers.flow.com/cadence)
```

**Review by turbolent — COMMENTED** [review]
Use full URLs instead

**Pull request merged**